app/instrumentation.py

```diff
@@ -288,8 +288,11 @@ def log_patch_group_success(logger, group_id):
     logger.info(f"Patched group: {group_id}")
 
 
-def log_patch_group_failed(logger, group_id):
-    logger.debug(f"Failed to find group during patch operation: {group_id}")
+def log_patch_group_failed(logger, group_id, message=None):
+    if not message:
+        logger.debug(f"Failed to find group during patch operation: {group_id}")
+    else:
+        logger.debug(f"Failed to patch group with id: {group_id} due to {message}")
```

Comment:
I'm not a fan of the log's intent changing based on whether a message was provided. Instead, you could just set the message's default value:
```suggestion
def log_patch_group_failed(logger, group_id, message="Group not found."):
    logger.debug(f"Failed to patch group with id {group_id}: {message}")
```

Reply:
Much better, thanks.